lib/ansible/modules/network/apconos/apconos_cert.py

```diff
@@ -0,0 +1,190 @@
+  filename:
+    description:
+      - specify a remote file name  which resides on a remote tftp server.
+  wait_for:
+    description:
+      - list of conditions to evaluate against the output of the
+        command. The task will wait for each condition to be true
+        before moving forward. If the conditional is not true
+        within the configured number of retries, the task fails.
+        See examples.
+  match:
+    description:
+      - The I(match) argument is used in conjunction with the
+        I(wait_for) argument to specify the match policy.  Valid
+        values are C(all) or C(any).  If the value is set to C(all)
+        then all conditionals in the wait_for must be satisfied.  If
+        the value is set to C(any) then only one of the values must be
+        satisfied.
+    default: all
+    choices: ['any', 'all']
+  retries:
+    description:
+      - Specifies the number of retries a command should by tried
+        before it is considered failed. The command is run on the
+        target device every retry and evaluated against the
+        I(wait_for) conditions.
+    default: 10
+  interval:
+    description:
+      - Configures the interval in seconds to wait between retries
+        of the command. If the command does not pass the specified
+        conditions, the interval indicates how long to wait before
+        trying the command again.
+    default: 1
+"""
+
+EXAMPLES = """
+- name: Install SSL Certificate
+  apconos_cert:
+    ipaddress: 10.0.0.100
+    filename:  remoteSSLCert.pem
+"""
+
+RETURN = """
+"""
+
+import re
+import time
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.network.common.parsing import Conditional
+from ansible.module_utils.network.common.utils import remove_default_spec
+from ansible.module_utils.network.apconos.apconos import load_config, run_commands
+from ansible.module_utils.network.apconos.apconos import apconos_argument_spec, check_args
+from ansible.module_utils.six import string_types
+
+
+def to_lines(stdout):
+    for item in stdout:
+        if isinstance(item, string_types):
+            item = str(item).split('\n')
+        yield item
+
+
+def construct_update_command(module):
+    """construct update command
+    """
+    command = module.params['command']
+    ipaddress = module.params['ipaddress']
+    filename = module.params['filename']
+    command[0] = 'tftp put ssl ipv4 ' + ipaddress[0] + ' ' + filename[0] + time.strftime("%d_%b+%H:%M:%S", time.gmtime())
+    command.append('tftp get ssl ipv4 ' + ipaddress[0] + ' ' + filename[0])
+
+    return command
+
+
+def main():
+    """ main entry point for module execution
+    """
+    spec = dict(
+        wait_for=dict(type='list'),
+        match=dict(default='all', choices=['all', 'any']),
+
+        retries=dict(default=10, type='int'),
+        interval=dict(default=1, type='int'),
+
+        ipaddress=dict(type='list'),
```

Comment:
Why would this be a list instead of a `str`? Also, other modules normally use camel case. I'll defer to a core developer to see what they think about this convention.

Reply:
str should be OK, but it is supposed to be a list for the application.

Reply:
What do you mean? How is this interfacing and what’s the application?

Reply:
The module is only using `ipaddress[0]`, so this shouldn't be a list.

Reply:
> 
> 
> The module is only using `ipaddress[0]`, so this shouldn't be a list.

I'll fix it.
Thanks!